WordPress/Classes/ViewRelated/Post/PostCardActionBar.m

```diff
@@ -0,0 +1,354 @@
+    button = [self.buttons firstObject];
+    views = NSDictionaryOfVariableBindings(button);
+    [self.contentView addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"|[button]"
+                                                                             options:0
+                                                                             metrics:nil
+                                                                               views:views]];
+    [self.contentView addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|[button]|"
+                                                                             options:0
+                                                                             metrics:nil
+                                                                               views:views]];
+
+
+    // in-between buttons
+    for (NSInteger i = 1; i < [self.buttons count] - 1; i++) {
+        previousButton = button;
+        button = [self.buttons objectAtIndex:i];
+        views = NSDictionaryOfVariableBindings(button, previousButton);
+        [self.contentView addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"[previousButton]-(1)-[button(==previousButton)]"
+                                                                                 options:0
+                                                                                 metrics:nil
+                                                                                   views:views]];
+        [self.contentView addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|[button]|"
+                                                                                 options:0
+                                                                                 metrics:nil
+                                                                                   views:views]];
+    }
+
+    previousButton = button;
+    button = [self.buttons lastObject];
+    views = NSDictionaryOfVariableBindings(button, previousButton);
+    [self.contentView addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"[previousButton]-(1)-[button(==previousButton)]|"
+                                                                             options:0
+                                                                             metrics:nil
+                                                                               views:views]];
+    [self.contentView addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|[button]|"
+                                                                             options:0
+                                                                             metrics:nil
+                                                                               views:views]];
+    [self setNeedsUpdateConstraints];
+}
+
+- (void)setupButtons
+{
+    for (UIButton *button in self.buttons) {
+        [button removeFromSuperview];
```

Comment:
Fun fact: removing all the buttons is slower than creating new buttons

![screen shot 2015-04-27 at 17 49 26](https://cloud.githubusercontent.com/assets/8739/7351484/c8f64c3e-ed05-11e4-9a95-680831705735.png)

I profiled this on an iPhone 5, and while it's sluggish at times, it feels responsive enough. I don't think it needs performance improvements, but I was surprised by this and wanted to share it :smile: